data/data-pipeline/data_pipeline/tests/score/test_output.py

```diff
@@ -203,3 +212,44 @@ def test_donut_hole_addition_to_score_n(final_score_df):
     assert (
         new_donuts > 0
     ), "FYI: The adjacency index is doing nothing. Consider removing it?"
+
+
+def test_data_sources(
+    final_score_df, hud_housing_df, ejscreen_df, cdc_places_df
+):
+    data_sources = {
+        key: value for key, value in locals().items() if key != "final_score_df"
+    }
+
+    for data_source_name, data_source in data_sources.items():
+        final = "_final"
+        df: pd.DataFrame = final_score_df.merge(
+            data_source,
+            on=GEOID_TRACT_FIELD_NAME,
+            indicator="MERGE",
+            suffixes=(final, f"_{data_source_name}"),
+            how="left",
+        )
+        data_source_columns = [
+            f"{col}_{data_source_name}"
+            for col in data_source.columns
+            if (col != GEOID_TRACT_FIELD_NAME and col in final_score_df.columns)
+        ]
+        final_columns = [
+            f"{col}{final}"
+            for col in data_source.columns
+            if (col != GEOID_TRACT_FIELD_NAME and col in final_score_df.columns)
+        ]
+        assert np.all(df[df.MERGE == "left_only"][final_columns].isna())
```

Comment:
this is nice. Another way to do the check is: 

```
df[df.MERGE == "left_only"][final_columns].notna().sum() == 0
```

That's not necessarily *better* but may read a bit more easily L -> R

Reply:
`isnull().all()` returns a series, so it ends up needing to be `isnull().all().all()` which felt slightly confusing to me. Since there's a builtin `all()` in python, I felt okay with this being idiomatic enough.